src/public/components/restricted-input.js

```diff
@@ -0,0 +1,43 @@
+import K, * as U from "karet.util"
+import React     from "karet"
+
+export const RestrictedInput = ({value, meta: {format, parse}, ...props}) => {
+  const edited = U.atom(undefined)
```

Comment:
I think I understand. You're saying you'd pass in `edited` so the caller could also be aware of changes to `edited` in case it needed to be, correct? Would you pass anything else in like `shown` and/or `validity`? And where would you put them in the args list? i.e. `({value, meta: {format, parse, edited}, ...props})` or what?